File: 1001-n-repeated-element-in-size-2n-array/n-repeated-element-in-size-2n-array.py
class Solution:
    def repeatedNTimes(self, nums: List[int]) -> int:
        n = len(nums) / 2
        freq = {}

        for i in nums:
            freq[i] = freq.get(i, 0) + 1
        
        for key, value in freq.items():
            if value == n:
                return key
        

File: 1001-n-repeated-element-in-size-2n-array/n-repeated-element-in-size-2n-array.swift
class Solution {
    func repeatedNTimes(_ nums: [Int]) -> Int {
        let target = nums.count / 2          // = n
        var counts = [Int: Int]()            // remainder: element → frequency
        
        for num in nums {
            let newCount = (counts[num] ?? 0) + 1
            if newCount == target {          // first time this element reaches n
                return num                   // answer guaranteed by problem
            }
            counts[num] = newCount
        }
        
        fatalError("Input constraints guarantee a repeated element") // safety net
    }
}